src/SimpleStubs.CodeGen/Utils/StubbingUtils.cs

```diff
@@ -1,13 +1,60 @@
+using System.Collections.Generic;
 using System.Linq;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SF = Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
 
 namespace Etg.SimpleStubs.CodeGen.Utils
 {
     internal class StubbingUtils
     {
-        public static string GenerateInvokeDelegateStmt(string delegateTypeName, string methodName, string parameters)
+        public static BlockSyntax GetInvocationBlockSyntax(string delegateTypeName, string methodName, string parameters, 
+            IEnumerable<IParameterSymbol> outParameters, bool returnsVoid, ITypeSymbol returnType, SemanticModel semanticModel)
```

Comment:
outParameters should not expect null but an empty IEnumerable when there are no out parameters (see Enumerable.Empty<>()). In fact, you can even leave it up to this method to find out parameters if needed.

Also do you need both returnsVoid and returnType? Can you deduce returnsVoid from returnType?